internal/pkg/client/ocisif/ocisif.go

```diff
@@ -202,12 +228,13 @@ func convertLayoutToOciSif(layoutDir string, digest v1.Hash, imageDest, workDir
 	return ocisif.Write(imageDest, ii)
 }
 
-// pushOCISIF pushes a single image from an OCI SIF to the OCI registry destination ref.
-func pushOCISIF(ctx context.Context, sourceFile, dest string, ociAuth *ocitypes.DockerAuthConfig) error {
-	dest = strings.TrimPrefix(dest, "//")
-	ref, err := name.ParseReference(dest)
+// PushOCISIF pushes a single image from sourceFile to the OCI registry destRef.
+func PushOCISIF(ctx context.Context, sourceFile, destRef string, ociAuth *ocitypes.DockerAuthConfig) error {
+	destRef = strings.TrimPrefix(destRef, "docker://")
+	destRef = strings.TrimPrefix(destRef, "//")
```

Comment:
I am not understanding how `destRef` could  be either `docker:////` or `//<something>`. Is line 234 necessary?

Reply:
Unfortunately there are differences between how higher-level code, across builds, the `pull` CLI, and the `actions` CLI send references to clients. This is a defence against that.

There has to be stripping of `docker://` to handle when the full ref is sent, and `//` for the case where the caller strips the `docker:` part, but not the following `//`.

Same for oras:

<img width="830" alt="image" src="https://github.com/sylabs/singularity/assets/4522799/300e8f64-7772-441d-862e-c6a7c9d20762">

I really want to tidy this up, but there are many paths here to rationalise... so I'll tackle it separately. Ultimately the aim is to switch to using only google/go-containerregistry, and remove the remaining duplicated code between build / pull / actions which complicates all of this.